pkg/controller/cronjob/utils.go

```diff
@@ -110,19 +110,21 @@ func getRecentUnmetScheduleTimes(cj batchv1.CronJob, now time.Time) ([]time.Time
 		// CronJob as last known start time.
 		earliestTime = cj.ObjectMeta.CreationTimestamp.Time
 	}
+
+	latestTime := now
 	if cj.Spec.StartingDeadlineSeconds != nil {
 		// Controller is not going to schedule anything below this point
-		schedulingDeadline := now.Add(-time.Second * time.Duration(*cj.Spec.StartingDeadlineSeconds))
+		schedulingDeadline := sched.Next(earliestTime).Add(time.Second * time.Duration(*cj.Spec.StartingDeadlineSeconds))
 
-		if schedulingDeadline.After(earliestTime) {
-			earliestTime = schedulingDeadline
+		if schedulingDeadline.Before(now) {
+			latestTime = schedulingDeadline
 		}
 	}
-	if earliestTime.After(now) {
+	if earliestTime.After(latestTime) {
```

Comment:
here logic may has some mistake, I think  the earliestTime must before then  latestTime,  whether the latestTime  from now or from sched.Next(earliestTime).Add(time.Second * time.Duration(*cj.Spec.StartingDeadlineSeconds)), so the if will never be true.

Reply:
There's something wrong with this line, so I've rewritten it. Thank you for your reminding.